lib/iris/analysis/cartography.py

```diff
@@ -1008,8 +1008,8 @@ def _transform_distance_vectors_tolerance_mask(
     u_one_t, v_zero_t = _transform_distance_vectors(ones, zeros, ds, dx2, dy2)
     u_zero_t, v_one_t = _transform_distance_vectors(zeros, ones, ds, dx2, dy2)
     # Squared magnitudes should be equal to one within acceptable tolerance.
-    # A value of atol=2e-3 is used, which corresponds to a change in magnitude
-    # of approximately 0.1%.
+    # A value of atol=2e-3 is used, which masks any magnitude changes >0.5%
+    #  (approx percentage - based on experimenting).
```

Comment:
Hi @trexfeathers and @bjlittle, I had come across the test failures in `test_rotate_winds.py` that were addressed in this pull request while working on issue #4972.

If I am not mistaken, the code essentially estimates the matrix norm `||A||` of a rotation matrix `A` that represents the actions of function `iris.analysis.cartography._transform_distance_vectors`, with
```
||A|| = inf/sup{ ||Ax|| for all ||x|| = 1}
```
where `||x||` is the magnitude of vector `x`, to find and mask cases with `||A|| < 0.999` or `||A|| > 1.001` where vector magnitudes are stretched or shrunken beyond a threshold of 0.1%.

At the moment, only two vector directions are tested, `x=(1,0)` and `x=(0,1)`, which doesn't seem to capture the matrix norm sufficiently to identify all cases that fall outside the threshold.

The situation improves when one adds additional directions using something like
```
    ones = np.ones(x.shape)
    mask = np.logical_and(ones,ones)
    for angle in np.linspace(0, 0.5*np.pi, num=4):
        u_t, v_t = _transform_distance_vectors(np.cos(angle)*ones, np.sin(angle)*ones, ds, dx2, dy2)
        mag_sq = u_t**2 + v_t**2
        mask = np.logical_and(
            mask,
            np.isclose(mag_sq, ones, atol=2e-3),
        )
    mask = np.logical_not(mask)
```
For the failing unit test, unmasked vectors are now very close to the 0.1% threshold. The downside is that it is evidently a bit more expensive to compute, and the number and range of angles are free parameters - 4 angles between 0 and 90 degrees work in this case, but it's not immediately clear that this will suffice in others.

Not sure how critical it is for rotated wind vector magnitudes to stay close to the stretching threshold, but just in case you are interested in looking at that.